### PR TITLE
improve `ShampineCollocationInit` initial dt choice

### PR DESCRIPTION
**Pull request by oscardssmith**
this makes the in place and OOP more aligned for `DAEProblem` and makes the `ODEProblem` `dt` timespan aware.

### CODE REVIEW

File: src/initialize_dae.jl [hunk 1]
```diff
@@ -225,7 +227,7 @@ function _initialize_dae!(integrator, prob::DAEProblem,
     tmp = get_tmp_cache(integrator)[1]
     resid = get_tmp_cache(integrator)[2]
 
-    dt = t != 0 ? min(t / 1000, dtmax) : dtmax # Haven't implemented norm reduction
+    dt = t != 0 ? min(t / 1000, dtmax / 10) : dtmax / 10 # Haven't implemented norm reduction
```
Comment (oscardssmith) [inline src/initialize_dae.jl:230]: @ChrisRackauckas is there a reason this isn't checking for `alg.initdt`?

**Comment by oscardssmith** [comment]
why does CI think things are failing? the logs don't show an error

**Comment by ChrisRackauckas** [comment]
Valentin turned off the MIT computers.

File: src/initialize_dae.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/initialize_dae.jl:230]: maybe it should

**Pull request merged**